parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetVectorizedRecordReader.java

```diff
@@ -1,251 +0,0 @@
-/**
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.parquet.hadoop;
-
-import static org.apache.parquet.filter2.compat.RowGroupFilter.filterRowGroups;
-import static org.apache.parquet.format.converter.ParquetMetadataConverter.NO_FILTER;
-import static org.apache.parquet.format.converter.ParquetMetadataConverter.range;
-import static org.apache.parquet.hadoop.ParquetFileReader.readFooter;
-import static org.apache.parquet.hadoop.ParquetInputFormat.getFilter;
-
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.fs.Path;
-import org.apache.hadoop.mapred.Reporter;
-import org.apache.hadoop.mapreduce.InputSplit;
-import org.apache.hadoop.mapreduce.RecordReader;
-import org.apache.hadoop.mapreduce.TaskAttemptContext;
-import org.apache.hadoop.mapreduce.TaskInputOutputContext;
-
-import org.apache.parquet.Log;
-import org.apache.parquet.column.ColumnDescriptor;
-import org.apache.parquet.filter.UnboundRecordFilter;
-import org.apache.parquet.filter2.compat.FilterCompat;
-import org.apache.parquet.filter2.compat.FilterCompat.Filter;
-import org.apache.parquet.filter2.compat.RowGroupFilter;
-import org.apache.parquet.hadoop.api.ReadSupport;
-import org.apache.parquet.hadoop.metadata.BlockMetaData;
-import org.apache.parquet.hadoop.metadata.ParquetMetadata;
-import org.apache.parquet.hadoop.util.ContextUtil;
-import org.apache.parquet.hadoop.util.counters.BenchmarkCounter;
-import org.apache.parquet.schema.MessageType;
-import org.apache.parquet.schema.MessageTypeParser;
-import org.apache.parquet.vector.ColumnVector;
-
-/**
- * Reads the records from a block of a Parquet file
- *
- * @see ParquetInputFormat
- *
- * @author Zhenxiao Luo
- *
- * @param <T> type of the materialized records
- */
-public class ParquetVectorizedRecordReader<T> extends RecordReader<Void, T>
```

Comment:
@dongche we remove `ParquetVectorizedRecordReader` with this PR. And yep it makes sense to also support batch reads from `ParquetRecordReader` too. Currently both  `ParquetReader` and `ParquetRecordReader` uses the `InternalParquetRecordReader` class to do the reads and `InternalParquetRecordReader` already has a `nextBatch()` method, which you can use at `ParquetRecordReader`. Please go ahead if you want to implement it.